### ISSUE TITLE
MapSnapshot issue with overlays

### ISSUE THREAD

**Issue by pasniak** [osmdroid/osmdroid#1450]
## Issue Type

[x] Question
[x] Bug?

## Description and/or steps/code to reproduce the problem

TLDR: overlays (4th arg of MapSnapshot) are not drawn/drawn in the bitmap in weird way. How do I make them draw "normally"?  

I create a [(MapSnapshot with overlay)]https://github.com/osmdroid/osmdroid/blob/97bc95e4338e8b4147f7cfa1e3245a53c4474ea7/osmdroid-android/src/main/java/org/osmdroid/views/drawing/MapSnapshot.java#L97) from a map location to draw tapped area on the map and it works **ok** when `overlays` argument is null. The map (zoomed in a bit*) appears in the bitmap.

When I add the my _custom_ overlays (which draw on canvas at the area, and which work well on the regular mapview i.e. they are drawn nicely at the tapped point), the custom overlays (which are actually in a FolderOverlay) do **not** appear in the bitmap. I checked and the FolderOverlay **is** in the passed list.

When I pass _all_ my mapView.overlays (I have a CompassOverlay for example), the only thing I can see is compass from CompassOverlay, which is drawn to high scale (even if it is far from the tapped point!).

*) I pass MapSnapshot.INCLUDE_FLAG_UPTODATE and zoom level whichis a bit smaller than the map (so I can see more detail about the point tapped)

### Version of osmdroid the issue relates to:

6.1.3

**Comment by monsieurtanuki** [comment]
@pasniak It's probably because in the overlays that don't appear in the snapshot you implemented draw(Canvas,MapView) (which used to be the correct and only way to draw overlays) but you should implement draw(Canvas,Projection) instead.
For snapshots we don't have a MapView, just a Projection.
Tell me if it works.

**Comment by pasniak** [comment]
That works! I added another draw() to my overlay and it is drawn. 

Note that SimpleFastPointOverlay (osmdroid-android/src/main/java/org/osmdroid/views/overlay/simplefastpoint/SimpleFastPointOverlay.java) has no draw(Canvas,Projection)
